src/data/set/finite.lean

```diff
@@ -347,6 +347,51 @@ by simp [set.ext_iff]
 @[simp] lemma coe_to_finset' [decidable_eq α] (s : set α) [fintype s] : (↑s.to_finset : set α) = s :=
 by ext; simp
 
+section preimage
+
+noncomputable def preimage {α β : Type*} {f : α → β} (s : finset β)
+  (hf : set.inj_on f (f ⁻¹' s.to_set)) : finset α :=
+set.finite.to_finset (set.finite_of_finite_image_on hf
+    (set.finite_subset s.finite_to_set (set.image_preimage_subset _ _)))
```

Comment:
This seems like it should be in `finset.lean`

Reply:
Unfortunately, `set.finite` is defined based on `finset`. Or do you see a way to make this definition without using `set.finite`?

Reply:
It doesn't need `set.finite`. You just need to map the finset with `inv_fun_on f`

Reply:
I didn't end up using `inv_fun_on`, but I've moved this definition.